src/inmanta/ast/__init__.py

```diff
@@ -143,6 +143,38 @@ def __eq__(self, other: object) -> bool:
         return False
 
 
+class AnchorTarget(Location):
+    __slots__ = ("start_char", "end_lnr", "end_char", "docstring")
+
+    def __init__(
+        self,
+        file: str,
+        start_lnr: int,
+        start_char: Optional[int],
+        end_lnr: Optional[int],
+        end_char: Optional[int],
+        docstring: Optional[str],
+    ) -> None:
+        """
+        Create a new AncherTarget instance.
+        :param file: the file this Range is in
+        :param start_lnr: the line number this Range starts on, 1-based
+        :param start_char: the start character number of the Range, 1-based
+        :param end_lnr: the line number this Range ends on, 1-based
+        :param end_char: the end character number of the Range, exclusive, 1-based
+        :param docstring: the docstring of the target
+        """
+        Location.__init__(self, file, start_lnr)
+        self.start_char = start_char
+        self.end_lnr = end_lnr
+        self.end_char = end_char
+        self.docstring = docstring
+
+
+class WithComment(object):
+    comment: str = ""
```

Comment:
Why not `Optional[str] = None`? I feel like this would better convey that there may not be a comment attached.

Reply:
For me it means the same `""` also means there is nothing attached as it impossible to have otherwise. and I don't like having Optionals everywhere in the code afterwards. But If you think it is wrong or not the "python" way of doing it I will change it :p

Reply:
It was just my first way of doing it, I'm not particularly attached to it. I'll see if I can change it